### PR TITLE
Add missing permission options for sftp adapter

### LINKED ISSUE #17: Cannot set "directoryPerm" option on sftp adapter
When add option to sftp adapter, its say:
`The option "directoryPerm" does not exist. Defined options are: "host", "password", "port", "private_key", "root", "timeout", "username".`

I think the problem is in [`\League\FlysystemBundle\Adapter\Builder\SftpAdapterDefinitionBuilder::configureOptions()`](https://github.com/thephpleague/flysystem-bundle/blob/cdd73c825031a5330f184f4ce73200211e426df2/src/Adapter/Builder/SftpAdapterDefinitionBuilder.php#L37-L59) because there is no `directoryPerm` set allow type to `OptionsResolver`

### PR DESCRIPTION
**Pull request by chaniwat**
Fixes #17

Add `directoryPerm`, `permPrivate` and `permPublic` to configurable (for directory/file permission), also add to the sftp document

### CODE REVIEW

File: docs/3-interacting-with-ftp-and-sftp-servers.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ flysystem:
                 private_key: 'path/to/or/contents/of/privatekey'
                 root: '/path/to/root'
                 timeout: 10
+                directoryPerm: 0744
```
Comment (tgalopin) [inline docs/3-interacting-with-ftp-and-sftp-servers.md]: Would be nice to put all the possible options here IMO.

File: src/Adapter/Builder/SftpAdapterDefinitionBuilder.php [hunk 2]
```diff
@@ -56,6 +56,15 @@ protected function configureOptions(OptionsResolver $resolver)
 
         $resolver->setDefault('timeout', 90);
         $resolver->setAllowedTypes('timeout', 'scalar');
+
+        $resolver->setDefault('directoryPerm', 0744);
+        $resolver->setAllowedTypes('directoryPerm', 'scalar');
+
+        $resolver->setDefault('permPrivate', 0700);
+        $resolver->setAllowedTypes('directoryPerm', 'scalar');
```
Comment (tgalopin) [inline src/Adapter/Builder/SftpAdapterDefinitionBuilder.php]: I think this should be `permPrivate`

File: src/Adapter/Builder/SftpAdapterDefinitionBuilder.php [hunk 3]
```diff
@@ -56,6 +56,15 @@ protected function configureOptions(OptionsResolver $resolver)
...
+
+        $resolver->setDefault('permPublic', 0744);
+        $resolver->setAllowedTypes('directoryPerm', 'scalar');
```
Comment (tgalopin) [inline src/Adapter/Builder/SftpAdapterDefinitionBuilder.php]: I think this should be `permPublic`

**Comment by chaniwat** [comment]
@tgalopin done :)

**Comment by tgalopin** [comment]
Thanks @chaniwat :) !

**Pull request merged**